examples/tv-app/android/java/ChannelManager.cpp

```diff
@@ -363,30 +371,273 @@ void ChannelManager::HandleGetProgramGuide(
-    // 2. Perform search
-    // 3. Return results
     ProgramGuideResponseType response;
-    // response.channelPagingStruct;
-    // response.programList;
-    helper.Success(response);
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+
+    ChipLogProgress(Zcl, "Received ChannelManager::HandleGetProgramGuide");
+    VerifyOrExit(mChannelManagerObject != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mGetProgramGuideMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != NULL, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        // NOTE: this example app does not pass the Data, PageToken, ChannelsArray or ExternalIdList through to the Java layer
+        UtfString jData(env, "");
+        UtfString jToken(env, "");
+        jobjectArray channelsArray       = nullptr;
+        jobjectArray externalIDListArray = (jobjectArray) env->NewObjectArray(0, env->FindClass("java/util/Map$Entry"), NULL);
+
+        jobject resp = env->CallObjectMethod(
+            mChannelManagerObject, mGetProgramGuideMethod, static_cast<jlong>(startTime.ValueOr(0)),
+            static_cast<jlong>(endTime.ValueOr(0)), channelsArray, jToken.jniValue(),
+            static_cast<jboolean>(recordingFlag.ValueOr(0).Raw() != 0), externalIDListArray, jData.jniValue());
+        if (env->ExceptionCheck())
+        {
+            ChipLogError(Zcl, "Java exception in ChannelManager::HandleGetProgramGuide");
+            env->ExceptionDescribe();
+            env->ExceptionClear();
+            err = CHIP_ERROR_INCORRECT_STATE;
+            goto exit;
+        }
+
+        VerifyOrExit(resp != nullptr, err = CHIP_JNI_ERROR_NULL_OBJECT);
+        jclass respCls       = env->GetObjectClass(resp);
+        jfieldID programsFid = env->GetFieldID(respCls, "programs", "[Lcom/matter/tv/server/tvapp/ChannelProgramInfo;");
+        VerifyOrExit(programsFid != nullptr, err = CHIP_JNI_ERROR_FIELD_NOT_FOUND);
+
+        std::vector<ProgramType *> needToFreePrograms;
+        std::vector<ChannelInfoType *> needToFreeChannels;
+        std::vector<ProgramType> tempPrograms;
+        std::vector<JniUtfString *> needToFreeStrings;
+
+        jobjectArray programsArray = (jobjectArray) env->GetObjectField(resp, programsFid);
+        jint size                  = env->GetArrayLength(programsArray);
+        for (jint i = 0; i < size; i++)
+        {
+            ProgramType * program = new ProgramType();
+            jobject programObject = env->GetObjectArrayElement(programsArray, i);
+            jclass programClass   = env->GetObjectClass(programObject);
+
+            jfieldID getIdentifierField = env->GetFieldID(programClass, "identifier", "Ljava/lang/String;");
+            jstring jidentifier         = static_cast<jstring>(env->GetObjectField(programObject, getIdentifierField));
+            JniUtfString * identifier   = new JniUtfString(env, jidentifier);
```

Comment:
check null and further release previous program if new fails

Reply:
fixed - thanks!